frontend/src/components/MapPopupContent/MapPopupContent.js

```diff
@@ -14,7 +18,42 @@ const StyledTitleLink = styled(Link)({
   marginBottom: '8px',
 })
 
-function MapPopupContent({ id, datasetId, title, metricName, data, colorScheme, domain, expanded }) {
+function MapPopupContent({
+  id,
+  datasetId,
+  title,
+  metricName,
+  data,
+  colorScheme,
+  domain,
+  expanded,
+  comparisonList,
+  addToComparison,
+}) {
+  const dispatch = useDispatch()
+
+  const disableAddToComparison =
+    comparisonList?.length >= MAX_COMPARE_COUNT || Boolean(comparisonList.find(item => item?.id === id))
+
+  const openCompareMenuOpen = useCallback(() => dispatch(setCompareMenuOpen(true)), [dispatch])
+
+  const handleAddToComparison = useCallback(() => {
+    dispatch(
+      getDatasetGeoJSON({
+        datasetId,
+        params: {
+          ids: id,
+          include_neighbors: true,
+          format: 'json',
+        },
+        success: geoJson => {
+          addToComparison({ id, title, bbox: geoJson?.bbox, features: geoJson?.features })
+          openCompareMenuOpen()
+        },
+      }),
+    )
+  }, [addToComparison, datasetId, dispatch, id, openCompareMenuOpen, title])
```

Comment:
Adding a feature shouldn't wait until the geometry endpoint has responded successfully. The request might fail or take a long time to respond.

Also, all we should really store here is `id` and `title`. I wouldn't store `bbox` or `features`. We don't know how large the geometry response will be, and some uses of the Atlas might involve feature geometry which changes often enough that we won't want to cache stale versions.